examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/endpoints/auth/GitHubHttpEndpoint.scala

```diff
@@ -0,0 +1,28 @@
+package com.example.http4s.blaze.demo.server.endpoints.auth
+
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.example.http4s.blaze.demo.server.endpoints.ApiVersion
+import com.example.http4s.blaze.demo.server.service.GitHubService
+import org.http4s._
+import org.http4s.dsl.Http4sDsl
+
+class GitHubHttpEndpoint[F[_]](gitHubService: GitHubService[F])
+                              (implicit F: Sync[F]) extends Http4sDsl[F] {
+
+  object CodeQuery extends QueryParamDecoderMatcher[String]("code")
+  object StateQuery extends QueryParamDecoderMatcher[String]("state")
```

Comment:
I have worked on http4s for a long time and almost never dealt with query parameters and I'm reminded every time that we need to make this better.  Did unfiltered have a better solution?  It's such a similar DSL.

Reply:
I have no idea of what you're talking about. Can you expand on "unfiltered"?

Reply:
[Unfiltered](https://github.com/unfiltered/unfiltered) is one of the first Scala HTTP libraries to be built on extractors.

Reply:
I'll look into it!